### PR TITLE
nwg-hello: init at 0.1.6

### LINKED ISSUE #279709: Package request: nwg-hello
GTK3-based greeter for greetd written in python 

**Metadata**

* homepage URL: https://github.com/nwg-piotr/nwg-hello
* source URL: https://github.com/nwg-piotr/nwg-hello
* license: MIT
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by khaneliman**
## Description of changes

Closes: https://github.com/NixOS/nixpkgs/issues/279709

Might try working on module right away to use it. 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/nw/nwg-hello/package.nix [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    substituteInPlace nwg_hello/main.py --replace '/var/cache/nwg-hello' $out/var/cache/nwg-hello
+  '';
+
+  postInstall = ''
+    install -D -m 644 -t $out/etc/nwg-hello/ nwg-hello-default.json
+    install -D -m 644 -t $out/etc/nwg-hello/ nwg-hello-default.css
+    install -D -m 644 -t $out/etc/nwg-hello/ hyprland.conf
+    install -D -m 644 -t $out/etc/nwg-hello/ sway-config
+    install -D -m 644 -t $out/etc/nwg-hello/ README
+    install -D -m 644 -t $out/usr/share/nwg-hello/ nwg.jpg
```
Comment (AndersonTorres) [inline pkgs/by-name/nw/nwg-hello/package.nix:60]: Do not use  `/usr`

File: pkgs/by-name/nw/nwg-hello/package.nix [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    install -D -m 644 -t $out/usr/share/nwg-hello/ img/*
+
+    # TODO: replace with temp file creation
+    install -d $out/var/cache/nwg-hello
+    install -Dm644 -t $out/var/cache/nwg-hello cache.json
+    # install -Dm644 -t $out/var/cache/nwg-hello cache.json -o greeter
```
Comment (AndersonTorres) [inline pkgs/by-name/nw/nwg-hello/package.nix:66]: Comment?

File: pkgs/by-name/nw/nwg-hello/package.nix [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+    # python3 setup.py install --optimize=1
+  '';
+
+  # preFixup = ''
+  #   makeWrapperArgs+=("''${gappsWrapperArgs[@]}");
```
Comment (AndersonTorres) [inline pkgs/by-name/nw/nwg-hello/package.nix:72]: Comment?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/nw/nwg-hello/package.nix (hunk 1 above)
Comment (khaneliman) [reply pkgs/by-name/nw/nwg-hello/package.nix]: just $out/share instead?

File: pkgs/by-name/nw/nwg-hello/package.nix [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  # '';
+
+  # Upstream has no tests
+  doCheck = false;
+
+  meta = {
+    homepage = "https://github.com/nwg-piotr/nwg-hello";
+    description = "";
```
[the remark this answers is not in the document]
Comment (khaneliman) [reply pkgs/by-name/nw/nwg-hello/package.nix]: sorry, my force push didn't go before i left my computer.

File: pkgs/by-name/nw/nwg-hello/package.nix [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+  # Upstream has no tests
+  doCheck = false;
+
+  meta = {
+    homepage = "https://github.com/nwg-piotr/nwg-hello";
+    description = "GTK3-based greeter for the greetd daemon, written in python";
+    license = lib.licenses.mit;
+    platforms = lib.platforms.linux;
+    maintainers = [ ];
```
Comment (GetPsyched) [inline pkgs/by-name/nw/nwg-hello/package.nix:67]: You're not going to be maintaining this package?

**Review by GetPsyched — APPROVED** [review]

**Comment by n3oney** [comment]
Sorry, missclicked!

File: pkgs/by-name/nw/nwg-hello/package.nix [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    substituteInPlace nwg-hello-default.json \
+      --replace "/usr/share/xsessions" "/run/current-system/sw/share/xsessions" \
+      --replace "/usr/share/wayland-sessions" "/run/current-system/sw/share/wayland-sessions"
+
+    substituteInPlace nwg_hello/ui.py --replace '/usr/share/nwg-hello' $out/share/nwg-hello
+  '';
+
+  postInstall = ''
+    install -D -m 644 -t $out/etc/nwg-hello/ nwg-hello-default.json
+    install -D -m 644 -t $out/etc/nwg-hello/ nwg-hello-default.css
```
Comment (Artturin) [inline pkgs/by-name/nw/nwg-hello/package.nix:51]:
```suggestion
    install -D -m 644 -t $out/etc/nwg-hello/ nwg-hello-default.json nwg-hello-default.css ...
```

`install` can take multiple SOURCE

File: pkgs/by-name/nw/nwg-hello/package.nix [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    substituteInPlace nwg_hello/main.py \
+      --replace '/etc/nwg-hello' $out/etc/nwg-hello \
+      --replace "/usr/share/xsessions" "/run/current-system/sw/share/xsessions" \
+      --replace "/usr/share/wayland-sessions" "/run/current-system/sw/share/wayland-sessions"
+
+    substituteInPlace nwg-hello-default.json \
+      --replace "/usr/share/xsessions" "/run/current-system/sw/share/xsessions" \
+      --replace "/usr/share/wayland-sessions" "/run/current-system/sw/share/wayland-sessions"
+
+    substituteInPlace nwg_hello/ui.py --replace '/usr/share/nwg-hello' $out/share/nwg-hello
```
Comment (Artturin) [inline pkgs/by-name/nw/nwg-hello/package.nix:46]:
```suggestion
    substituteInPlace nwg_hello/ui.py --replace '/usr/share/nwg-hello' "$out/share/nwg-hello"
```

File: pkgs/by-name/nw/nwg-hello/package.nix [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+  propagatedBuildInputs = [
+    gtk-layer-shell
+    python3Packages.pygobject3
+  ];
+
+  postPatch = ''
+    # hard coded paths
+    substituteInPlace nwg_hello/main.py \
+      --replace '/etc/nwg-hello' $out/etc/nwg-hello \
```
Comment (Artturin) [inline pkgs/by-name/nw/nwg-hello/package.nix:38]:
```suggestion
      --replace '/etc/nwg-hello' "$out/etc/nwg-hello" \
```

File: pkgs/by-name/nw/nwg-hello/package.nix [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    gobject-introspection
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    gtk3
+  ];
+
+  propagatedBuildInputs = [
+    gtk-layer-shell
```
Comment (Artturin) [inline pkgs/by-name/nw/nwg-hello/package.nix:31]: Why propagate

File: pkgs/by-name/nw/nwg-hello/package.nix [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    install -D -m 644 -t $out/etc/nwg-hello/ nwg-hello-default.css
+    install -D -m 644 -t $out/etc/nwg-hello/ hyprland.conf
+    install -D -m 644 -t $out/etc/nwg-hello/ sway-config
+    install -D -m 644 -t $out/etc/nwg-hello/ README
+    install -D -m 644 -t $out/share/nwg-hello/ nwg.jpg
+    install -D -m 644 -t $out/share/nwg-hello/ img/*
+  '';
+
+  # Upstream has no tests
+  doCheck = false;
```
Comment (Artturin) [inline pkgs/by-name/nw/nwg-hello/package.nix:60]: Add a `pythonImportsCheck`

**Review by Artturin — CHANGES_REQUESTED** [review]

**Review by n3oney — APPROVED** [review]

**Review by Artturin — APPROVED** [review]

**Pull request merged**